### PR TITLE
feat: upgrade `@sentry/react-native`

### LINKED ISSUE #293: Bump underlying Sentry JS dependency?
### Summary

Long-story short, `sentry-expo` requires version 7.7 of Sentry JS. We want to bump to ~7.28. (see [here](https://github.com/getsentry/sentry-react-native/blob/09db9f0ea4406ce884c9002eb43c887f600846a7/RNSentry.podspec#L20)) Because of this constraint, the `@sentry/react-native` library (needed for native reporting) is constrained to be at version`3.2` or lower. (iOS production builds will fail due to Podfile dep issues). `@sentry/react-native` is currently at `4.6.1` and version `5.0.0-alpha.6` is already out.

I believe there are some issues with the version of `@sentry/react-native`. In Android 6, 7 and 8, for example, any app containing it will simply not open. This is due to an error in this version of Sentry. I see multiple crash reports in Android Vitals about this. Android 7 is 5% of global users.

```java
Exception java.lang.RuntimeException: Could not invoke RNSentry.initNativeSdk
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:383)
  at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:188)
  at com.facebook.react.bridge.queue.NativeRunnable.run
  at android.os.Handler.handleCallback (Handler.java:836)
  at android.os.Handler.dispatchMessage (Handler.java:103)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
  at android.os.Looper.loop (Looper.java:203)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:228)
  at java.lang.Thread.run (Thread.java:761)
Caused by java.lang.reflect.InvocationTargetException:
  at java.lang.reflect.Method.invoke
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
Caused by java.lang.IncompatibleClassChangeError:
  at io.sentry.react.RNSentryModule.lambda$initNativeSdk$1$io-sentry-react-RNSentryModule (RNSentryModule.java:192)
  at io.sentry.react.RNSentryModule$$ExternalSyntheticLambda6.configure
  at io.sentry.android.core.SentryAndroid.lambda$init$1 (SentryAndroid.java:94)
  at io.sentry.android.core.SentryAndroid$$ExternalSyntheticLambda0.configure
  at io.sentry.Sentry.init (Sentry.java:116)
  at io.sentry.android.core.SentryAndroid.init (SentryAndroid.java:83)
  at io.sentry.android.core.SentryAndroid.init (SentryAndroid.java:64)
  at io.sentry.react.RNSentryModule.initNativeSdk (RNSentryModule.java:88)
```

I would assume (or blindly hope) that these issues are resolved in later versions of `@sentry/react-native`, therefore it might be worth bumping the version in this repo to allow for that to be installed. 

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

44

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.4
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
      Yarn: 1.22.18 - /opt/homebrew/bin/yarn
      npm: 8.11.0 - ~/.nvm/versions/node/v16.15.1/bin/npm
      Watchman: 2022.10.03.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
      Android SDK:
        API Levels: 30, 31
        Build Tools: 29.0.2, 30.0.2, 31.0.0, 33.0.0
        System Images: android-24 | Google APIs ARM 64 v8a, android-31 | Google APIs ARM 64 v8a
    IDEs:
      Android Studio: 2021.1 AI-211.7628.21.2111.8193401
      Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
    npmPackages:
      @expo/metro-config: ^0.3.9 => 0.3.21
      expo: ^45.0.0 => 45.0.8
      react: 17.0.2 => 17.0.2
      react-dom: 17.0.2 => 17.0.2
      react-native: 0.68.2 => 0.68.2
      react-native-web: 0.17.7 => 0.17.7
    npmGlobalPackages:
      eas-cli: 0.55.1
      expo-cli: 6.0.5
    Expo Workflow: managed

### Reproducible demo or steps to reproduce from a blank project

package.json

```json
{
  "dependencies": {
      "@sentry/react-native": "^3.1.1",
      "sentry-expo": "^4.2.0",
  }
}
```

### PR DESCRIPTION
**Pull request by SimenB**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/sentry-expo/blob/main/CHANGELOG.md#master) if necessary.

# Why

The current version is quite outdated.

Closes #293

# How

Upgraded all `@sentry` packages, and made the necessary type change.

I'm not sure if this should be considered a breaking change?

# Test Plan

Not sure - hopefully you have some test app you can use? 😀

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 ### 🎉 New features
 
+- Upgrade `@sentry/react-natuve` to `4.8.0` ([#301](https://github.com/expo/sentry-expo/pull/301) by [@SimenB](https://github.com/SimenB))
```
Comment (asharamseervi) [inline CHANGELOG.md:9]: Typo: `@sentry/react-native`

**Comment by kbrandwijk** [comment]
Would this be breaking because Sentry now expects these fields to be either set or `undefined`, and `expo-device` returns `null`?

**Comment by SimenB** [comment]
I don't think so - I assume the type is stricter than the implementation 🙂 Haven't checked tho.

The reason I think this _might_ be breaking is simply because the user's version must match

**Comment by kbrandwijk** [comment]
Thank you very much for your contribution. I've wrapped these changes into #302, as those are also breaking changes and I'd rather ship a single major release.

**Review by kbrandwijk — APPROVED** [review]

**Comment by SimenB** [comment]
Thanks!

**Comment by kbrandwijk** [comment]
@SimenB If you restore the other 2 branches, I'd be happy to do the same there.

**Comment by SimenB** [comment]
Done 👍 I'm happy to rebase if needed (I assume they conflict with this PR)

**Comment by kbrandwijk** [comment]
It doesn't let me reopen the PRs 
image
Could I bother you to put up 2 new PRs? You can just reference the old ones for details. Sorry for the inconvenience.

**Comment by SimenB** [comment]
hah, should have held off on the rebase 😛

**Pull request merged**